packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,252 @@
+  }
+}
+
+/**
+ * Properties of provisioning artifacts
+ */
+export interface ProvisioningArtifactProperties {
+
+  /**
+   * The description of the provisioning artifact
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Disable template validation.
+   * @default True
+   */
+  readonly disableTemplateValidation?: boolean;
+
+  /**
+   * The S3 url that points to the provisioning artifacts template
+   */
+  readonly templateUrl: string;
+
+  /**
+   * The name of the provisioning artifact.
+   * @default - No provisioning artifact name provided
+   */
+  readonly name?: string;
+}
+
+/**
+ * Properties for a Cloudformation Product
+ */
+export interface ProductProps {
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the product.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * The distributor of the product.
+   * @default - No distributor provided
+   */
+  readonly distributor?: string;
+
+  /**
+   * The owner of the product.
+   */
+  readonly owner: string;
+
+  /**
+   * The configuration of the provisioning artifact (also known as a version).
+   */
+  readonly provisioningArtifacts: ProvisioningArtifactProperties[];
+
+  /**
+   * ReplaceProvisioningArtifacts
+   * @default false
+   */
+  readonly replaceProvisioningParameters?: boolean;
+
+  /**
+   * The support information about the product
+   * @default - No support description provided
+   */
+  readonly supportDescription?: string;
+
+  /**
+   * The contact email for product support.
+   * @default - No support email provided
+   */
+  readonly supportEmail?: string;
+
+  /**
+   * The contact URL for product support.
+   * @default - No support URL provided
+   */
+  readonly supportUrl?: string;
+
+  /**
+   * One or more tags.
+   * @default - No tags provided
+   */
+  readonly tags?: Tag[];
+}
+
+/**
+ * A Service Catalog Cloudformation Product.
+ * @resource AWS::ServiceCatalog::CloudFormationProduct
+ */
+export class Product extends ProductBase {
+
+  /**
+   * Creates a Product construct that represents an external product.
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs Product import properties
+   */
+  public static fromProductAttributes(scope: Construct, id: string, attrs: ProductAttributes): IProduct {
+    const parts = Stack.of(scope).parseArn(attrs.productArn);
+    function throwError(errorMessage: string): never {
+      throw new Error(errorMessage);
+    }
+    class Import extends ProductBase {
+      public readonly productId = parts.resourceName ?? throwError('Product arn missing Product ID during import from attributes');
```

Comment:
Same comment here (include the value causing the error in the message):

```suggestion
      public readonly productId = parts.resourceName ?? throwError('Product ARN is missing Product ID: ' + attrs.productArn);
```